sound/soc/sof/probe.c

```diff
@@ -0,0 +1,205 @@
+// SPDX-License-Identifier: (GPL-2.0 OR BSD-3-Clause)
+//
+// This file is provided under a dual BSD/GPLv2 license.  When using or
+// redistributing this file, you may do so under either license.
+//
+// Copyright(c) 2019 Intel Corporation. All rights reserved.
+//
+// Author: Cezary Rojewski <cezary.rojewski@intel.com>
+//
+
+#include "sof-priv.h"
+#include "probe.h"
+
+int sof_ipc_probe_init(struct snd_sof_dev *sdev,
+		size_t buffer_size)
+{
+	struct sof_ipc_probe_dma_set_params *msg;
+	struct sof_ipc_reply reply;
+	unsigned int stream_tag = INVALID_NODE_ID;
+	size_t size;
+	int ret;
+
+	size = struct_size(msg, dma, 1);
+	msg = kzalloc(size, GFP_KERNEL);
+	if (!msg)
+		return -ENOMEM;
+
+	if (sdev->extractor)
+		stream_tag = sdev->extractor;
+
+	msg->hdr.size = size;
+	msg->hdr.cmd = SOF_IPC_GLB_PROBE | SOF_IPC_PROBE_INIT;
+	msg->dma[0].stream_tag = stream_tag;
```

Comment:
...or even remove `stream_tag` completely and calculate it directly here.

Reply:
Answer has been provided by me above.

Reply:
Aligned with your suggestion in v5.